### PR TITLE
Refactor client and server

### PR DESCRIPTION
**Pull request by SchumacherFM**
This is a breaking API change.

- removes unnecessary allocations
- Removes global variables
- Allow custom http.Client
- Make code testable
- add unit tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ import (
 	"encoding/xml"
 	"log"
 
-	"github.com/foomo/soap"
+	"github.com/globusdigital/soap"
```
Comment (loeffert) [inline README.md]: that should be github.com/foomo/soap in the docs

File: examples/client.go [hunk 2]
```diff
@@ -4,7 +4,7 @@ import (
 	"encoding/xml"
 	"log"
 
-	"github.com/foomo/soap"
+	"github.com/globusdigital/soap"
```
Comment (loeffert) [inline examples/client.go]: that should import github.com/foomo/soap

File: examples/simple-server.go [hunk 3]
```diff
@@ -3,9 +3,10 @@ package main
 import (
 	"encoding/xml"
 	"fmt"
+	"log"
 	"net/http"
 
-	"github.com/foomo/soap"
+	"github.com/globusdigital/soap"
```
Comment (loeffert) [inline examples/simple-server.go]: that should import github.com/foomo/soap

File: go.mod [hunk 4]
```diff
@@ -0,0 +1,5 @@
+module github.com/globusdigital/soap
```
Comment (loeffert) [inline go.mod]: that should be github.com/foomo/soap

**Review by loeffert — CHANGES_REQUESTED** [review]

File: client.go [hunk 5]
```diff
@@ -4,49 +4,35 @@ import (
...
 	"strings"
-	"time"
 )
 
-// ClientDialTimeout default timeout 30s
-var ClientDialTimeout = time.Duration(30 * time.Second)
-
 // UserAgent is the default user agent
-var UserAgent = "go-soap-0.1"
+const UserAgent = "go-soap-0.1"
```
Comment (janhalfar) [inline client.go]: the var here was actually intentional and I think it makes sense to allow users of the client to set their own user agent

**Review by janhalfar — COMMENTED** [review]
thank you for the great work, what to you think about the var UserAgent ?

File: client.go (hunk 5 above)
Comment (SchumacherFM) [reply client.go]: Fixed

**Comment by SchumacherFM** [comment]
@janhalfar  I don't like global variable, they often lead to weird problems. refactored your suggestion and now it's awesome ;-) please review

**Pull request closed without merging**